File: IntelligentRecruitment_microservice/application-service/src/main/java/com/fedicode/applicationservice/Dto/RecruiterApplicationStatsResponse.java
package com.fedicode.applicationservice.Dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecruiterApplicationStatsResponse {
    private long totalApplications;
    private long pendingReview;
    private long acceptedThisMonth;
    private long refusedThisMonth;
    private long applicationsThisWeek;
    private long applicationsPreviousWeek;
    private double averageScore;
    private Map<String, Long> weeklyData;
}

File: IntelligentRecruitment-UI/src/app/models/Dto/Application/RecruiterApplicationStatsResponse.ts

export interface RecruiterApplicationStatsResponse {
  totalApplications: number;
  pendingReview: number;
  acceptedThisMonth: number;
  refusedThisMonth: number;
  applicationsThisWeek: number;
  applicationsPreviousWeek: number;
  averageScore: number;
  weeklyData: { [day: string]: number };
}
